asciidoctorj-pdf/src/test/java/org/asciidoctor/util/pdf/ImageProcessor.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.pdfbox.util.Matrix;
+import org.apache.pdfbox.util.PDFOperator;
+import org.apache.pdfbox.util.PDFStreamEngine;
+import org.apache.pdfbox.util.ResourceLoader;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Processes a PDF document to extract the metadata of the contained images.
+ *
+ * Based on https://svn.apache.org/viewvc/pdfbox/tags/1.8.9/examples/src/main/java/org/apache/pdfbox/examples/util/PrintImageLocations.java
+ *
+ * @author abelsromero
+ */
+public class ImageProcessor extends PDFStreamEngine {
+    
+    private static final String INVOKE_OPERATOR = "Do";
+
+    private int currentPage = 0;
+    private List<Image> images = new ArrayList<Image>();
+
+    /**
+     * Default constructor
+     *
+     * @throws IOException If there is an error loading text stripper properties.
+     */
+    public ImageProcessor() throws IOException {
+        super(ResourceLoader.loadProperties(
+                "org/apache/pdfbox/resources/PDFTextStripper.properties", true));
+    }
+
+    /**
+     * Parses a document extracting the images
+     *
+     * @param filename PDF document path
+     */
+    public void parse(String filename) throws IOException {
+        PDDocument document = null;
+        try {
+            document = PDDocument.load(filename, false);
+            List allPages = document.getDocumentCatalog().getAllPages();
+            for( int i=0; i<allPages.size(); i++ ) {
+                PDPage page = (PDPage)allPages.get( i );
+                currentPage = i;
+                processStream( page, page.findResources(), page.getContents().getStream() );
+            }
+        } finally {
+            if (document != null) {
+                document.close();
+            }
+        }
+    }
+
+    /**
+     * This is used to handle an operation
+     *
+     * @param operator The operation to perform
+     * @param arguments The list of arguments
+     *
+     * @throws IOException If there is an error processing the operation.
+     */
+    protected void processOperator( PDFOperator operator, List arguments ) throws IOException {
+        String operation = operator.getOperation();
+
+        if (INVOKE_OPERATOR.equals(operation)) {
+            COSName objectName = (COSName)arguments.get( 0 );
+            Map<String, PDXObject> xobjects = getResources().getXObjects();
+            PDXObject xobject = (PDXObject)xobjects.get( objectName.getName() );
+
+            if (xobject instanceof PDXObjectImage)  {
+                PDXObjectImage image = (PDXObjectImage)xobject;
+                int imageWidth = image.getWidth();
+                int imageHeight = image.getHeight();
+                PDPage page = getCurrentPage();
+                double pageHeight = page.getMediaBox().getHeight();
+
+                Matrix ctmNew = getGraphicsState().getCurrentTransformationMatrix();
+                float yScaling = ctmNew.getYScale();
```

Comment:
I wonder if all these computations could be removed, as the test only counts the number of images?